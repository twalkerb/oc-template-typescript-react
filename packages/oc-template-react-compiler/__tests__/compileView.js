jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000;

const path = require("path");
const fs = require("fs-extra");
const compileView = require("../lib/compileView.js");

const componentPath = path.join(__dirname, "../../mocks/react-component");
const publishPath = path.join(componentPath, "_package");
const publishFileName = "template.js";
const componentPackage = require(componentPath + "/package.json");
const options = {
  componentPackage,
  componentPath,
  publishPath,
  publishFileName
};
// console.log(componentPackage.oc.files.template.src)
test("Should correctly compile the view", done => {
  compileView(options, (err, compiledViewInfo) => {
    delete compiledViewInfo.hashKey;
    expect(compiledViewInfo).toMatchSnapshot();
    // expect(
    //   fs.readFileSync(path.join(publishPath, publishFileName), 'UTF8')
    // ).toMatchSnapshot();
    // fs.removeSync(path.join(publishPath, publishFileName));
    done();
  });
});
