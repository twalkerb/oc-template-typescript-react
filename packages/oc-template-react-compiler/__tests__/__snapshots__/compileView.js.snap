// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valid component 1`] = `
Object {
  "bundle": Object {
    "hashKey": "f5f4960195f5274e0193ac0d4dd2d1339d7474f1",
  },
  "template": Object {
    "hashKey": "2bffee35969f58acfab98f4840e63c2ef638104b",
    "src": "template.js",
    "type": "oc-template-react",
  },
}
`;

exports[`valid component 2`] = `"var oc=oc||{};oc.components=oc.components||{};oc.components['2bffee35969f58acfab98f4840e63c2ef638104b']=function(model){ var modelHTML = model.__html ? model.__html : ''; var staticPath = model.reactComponent.props._staticPath; var props = JSON.stringify(model.reactComponent.props); return '<div id=\\"oc-reactRoot-react-component\\" class=\\"oc-reactRoot-react-component\\">' + modelHTML + '</div>' + '<script>' + 'window.oc = window.oc || {};' + 'oc.cmd = oc.cmd || [];' + 'oc.cmd.push(function(oc){' + 'oc.addStylesToHead(\\"null\\");' + 'oc.requireSeries([{\\"global\\":\\"PropTypes\\",\\"url\\":\\"https://unpkg.com/prop-types@15.6.0/prop-types.min.js\\",\\"name\\":\\"prop-types\\"},{\\"global\\":\\"React\\",\\"url\\":\\"https://unpkg.com/react@16.2.0/umd/react.production.min.js\\",\\"name\\":\\"react\\"},{\\"global\\":\\"ReactDOM\\",\\"url\\":\\"https://unpkg.com/react-dom@16.2.0/umd/react-dom.production.min.js\\",\\"name\\":\\"react-dom\\"}], function(){' + 'oc.require(' + '[\\"oc\\", \\"reactComponents\\", \\"f5f4960195f5274e0193ac0d4dd2d1339d7474f1\\"],' + '\\"' + staticPath + 'react-component.js\\",' + 'function(ReactComponent){' + 'var targetNode = document.getElementById(\\"oc-reactRoot-react-component\\");' + 'targetNode.setAttribute(\\"id\\",\\"\\");' + 'ReactDOM.render(React.createElement(ReactComponent,' + props + '),targetNode);' + '}' + ');' + '});' + '});' + '</script>' }"`;
